### PR TITLE
DE43762: New HTML editor triggers Discard Changes popup unnecessarily

### PR DESCRIPTION
**Pull request by coffeexcode**
## Relevant Rally Links
- [DE43762](https://rally1.rallydev.com/#/289692574792d/custom/355050439968?detail=%2Fdefect%2F600609760039&fdp=true?fdp=true): FACE  > Modules & HTML files > New HTML editor triggers Discard Changes popup unnecessarily

## Description
The new html editor automatically appends the head and body tags when it is focused. When creating a new HTML file, the persisted content will be empty, but by clicking into the new HTML editor the current content will contain the HTML structure.  Despite no content actually being in the editor, when discarding, the warning dialog will show unnecessarily.

There was a related PR https://github.com/BrightspaceHypermediaComponents/activities/pull/1825 that fixed the modules part of this error, so this PR only fixes the error for HTML files.

## Goal / Solution
When checking if the content file is dirty, we can also check the persisted file content. When the persisted file content is empty, it means the file is brand new (as saving an empty html file still has the surrounding HTML structure). This should be faster/simpler than stripping the added HTML structure from the current content.

### CODE REVIEW

File: components/d2l-activity-editor/d2l-activity-content-editor/content-file/state/content-file.js [hunk 1]
```diff
@@ -118,6 +118,18 @@ export class ContentFile {
 		return new ContentFileEntity(sirenEntity, this.token, { remove: () => { } });
 	}
 
+	_contentEquals() {
```
Comment (brissons) [inline components/d2l-activity-editor/d2l-activity-content-editor/content-file/state/content-file.js:121]: Up to you, but we could probably make this a one liner with a ternary.

**Review by brissons — APPROVED** [review]
LGTM! Did a quick dev test and works great.

**Review by ChrisKraljevicD2L — DISMISSED** [review]

**Review by ianlaidlaw — APPROVED** [review]

**Review by prajapac — DISMISSED** [review]
LGTM, Scott's ternary suggestion is neat.

File: components/d2l-activity-editor/d2l-activity-content-editor/content-file/state/content-file.js (hunk 1 above)
Comment (coffeexcode) [reply components/d2l-activity-editor/d2l-activity-content-editor/content-file/state/content-file.js:129]: Addressed 6195d3

File: components/d2l-activity-editor/d2l-activity-content-editor/content-file/d2l-activity-content-file-detail.js [hunk 2]
```diff
@@ -251,16 +251,6 @@ class ContentFileDetail extends SkeletonMixin(ErrorHandlingMixin(LocalizeActivit
 		this.replacementContent = null;
 	}
 
-	_isEditorEmpty() {
```
Comment (coffeexcode) [inline components/d2l-activity-editor/d2l-activity-content-editor/content-file/d2l-activity-content-file-detail.js:254]: Moving this onto content file to reduce duplication.

File: components/d2l-activity-editor/d2l-activity-content-editor/content-file/state/content-file.js [hunk 3]
```diff
@@ -26,7 +26,15 @@ export class ContentFile {
 	}
 
 	get dirty() {
-		return !(this._contentFileEntity.equals(this._makeContentFileData()) && this.persistedFileContent === this.fileContent);
+		return !(this._contentFileEntity.equals(this._makeContentFileData()) && this._contentEquals());
+	}
+
+	get empty() {
```
Comment (coffeexcode) [inline components/d2l-activity-editor/d2l-activity-content-editor/content-file/state/content-file.js:32]: The regex was taken from lessons directly.

File: components/d2l-activity-editor/d2l-activity-content-editor/content-file/state/content-file.js [hunk 4]
```diff
@@ -118,6 +126,18 @@ export class ContentFile {
 		return new ContentFileEntity(sirenEntity, this.token, { remove: () => { } });
 	}
 
+	_contentEquals() {
+		/* This check stops the discard dialog from appearing when no content
+			is added to the editor but it was clicked in. Faster than stripping
+			the html, body, etc. tags added by the new html editor
+		*/
+		if (this.persistedFileContent === '' && this.empty) {
```
Comment (coffeexcode) [inline components/d2l-activity-editor/d2l-activity-content-editor/content-file/state/content-file.js:134]: I originally missed the empty case, so if you typed something and the file wasn't saved it wouldn't warn you about losing content (very bad) but it's fixed now.

**Review by brissons — APPROVED** [review]

**Pull request merged**